components/patient-measurements/api/src/test/java/org/phenotips/measurements/internal/MeasurementAgeUpdaterTest.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+package org.phenotips.measurements.internal;
+
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Matchers;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.xwiki.component.manager.ComponentLookupException;
+import org.xwiki.component.util.ReflectionUtils;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.test.mockito.MockitoComponentMockingRule;
+import org.xwiki.observation.event.Event;
+
+import java.lang.reflect.Field;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.LinkedList;
+import java.util.List;
+
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.eq;
+
+
+/**
+ * Tests for the {@link MeasurementAgeUpdater} component.
+ *
+ */
+public class MeasurementAgeUpdaterTest
+{
+
+    @Rule
+    public MockitoComponentMockingRule<MeasurementAgeUpdater> mocker =
+            new MockitoComponentMockingRule<>(MeasurementAgeUpdater.class);
+
+    @Mock
+    private Object data;
+
+    @Mock
+    public XWikiDocument source;
+
+    @Mock
+    private Event event;
+
+    @Mock
+    private BaseObject patientRecordObj;
+
+    @Mock
+    private BaseObject measurement;
+
+    private String DATE_PROPERTY_NAME;
+
+    private String AGE_PROPERTY_NAME;
+
+    private List<BaseObject> objects;
+
+    @Before
+    public void setUp() throws NoSuchFieldException, IllegalAccessException
+    {
+        MockitoAnnotations.initMocks(this);
+        when(this.source.getXObject((EntityReference) Matchers.any())).thenReturn(this.patientRecordObj);
+
+        Field age = ReflectionUtils.getField(MeasurementAgeUpdater.class, "AGE_PROPERTY_NAME");
+        Field date = ReflectionUtils.getField(MeasurementAgeUpdater.class, "DATE_PROPERTY_NAME");
+
+        age.setAccessible(true);
+        date.setAccessible(true);
+
+        this.DATE_PROPERTY_NAME = (String)date.get(null);
```

Comment:
That's really convoluted. A better test would be to just define it here again, and check that the same string is indeed used.